### ISSUE TITLE
Dynamic walker exception at initSystem

### ISSUE THREAD

**Issue by jimmyDunne** [opensim-org/opensim-core#1388]
The dynamic walker model (DynamicWalker.osim') can be read into memory, but an exception is called at initsystem.

```
import org.opensim.modeling.*

model = Model('../Model/WalkerModel.osim');
```

```
Updating Model file from 30000 to latest format...
Body 'Pelvis_massless' is massless but nonzero inertia provided. Inertia reset to zero. Otherwise provide nonzero mass.
Loaded model WalkerModel from file ../Model/WalkerModel.osim
```

```
state = model.initSystem();
```

```
Body 'LeftThigh_slave_0' not connected by a Joint.
A FreeJoint will be added to connect it to ground.
Body 'RightThigh_slave_0' not connected by a Joint.
A FreeJoint will be added to connect it to ground.
Java exception occurred:
java.io.IOException: Assigned an invalid SimTK::MobilizedBodyIndex
	In object 'Platform_offset' of type PhysicalOffsetFrame.
	In file
        /Users/jimmy/repository/opensim-master/opensim-core/OpenSim/Simulation/Model/PhysicalFrame.cpp:73
        	In function 'setMobilizedBodyIndex'
	at
        org.opensim.modeling.opensimModelSimulationJNI.Model_initSystem(Native
        Method)
	at org.opensim.modeling.Model.initSystem(Model.java:875)
```

**Comment by jimmyDunne** [comment]
I am able to get the 'new' model working when I delete the joints RightThigh_slave_0 and LeftThigh_slave_0. These were created by the internal converter and don't seem to do anything.

**Comment by jimmyDunne** [comment]
The below joint type gets added to the model. 

```xml
<FreeJoint name="free_LeftThigh_slave_0">
					<!--List of connectors (structural dependencies) that this component has.-->
					<connectors>
						<Connector_PhysicalFrame_ name="parent_frame">
							<!--Name of the component this Connector should be connected to.-->
							<connectee_name>../ground</connectee_name>
						</Connector_PhysicalFrame_>
						<Connector_PhysicalFrame_ name="child_frame">
							<!--Name of the component this Connector should be connected to.-->
							<connectee_name>../LeftThigh/LeftThigh_slave_0</connectee_name>
						</Connector_PhysicalFrame_>
					</connectors>
					<!--List containing the generalized coordinates (q's) that parameterize this joint.-->
					<coordinates>
						<Coordinate name="free_LeftThigh_slave_0_coord_0">
							<!--All properties of this object have their default values.-->
						</Coordinate>
						<Coordinate name="free_LeftThigh_slave_0_coord_1">
							<!--All properties of this object have their default values.-->
						</Coordinate>
						<Coordinate name="free_LeftThigh_slave_0_coord_2">
							<!--All properties of this object have their default values.-->
						</Coordinate>
						<Coordinate name="free_LeftThigh_slave_0_coord_3">
							<!--All properties of this object have their default values.-->
						</Coordinate>
						<Coordinate name="free_LeftThigh_slave_0_coord_4">
							<!--All properties of this object have their default values.-->
						</Coordinate>
						<Coordinate name="free_LeftThigh_slave_0_coord_5">
							<!--All properties of this object have their default values.-->
						</Coordinate>
					</coordinates>
					<!--Advanced option. Specify the direction of the joint in the multibody tree: parent->child (forward, reverse is false) or child->parent (reverse is true) NOTE: the Joint transform and its coordinates maintain a parent->child sense, even if the Joint is reversed.-->
					<reverse>false</reverse>
				</FreeJoint>
```

**Comment by aseth1** [comment]
@jimmyDunne can you provide a link to the model. Thanks!

**Comment by tkuchida** [comment]
There's a .zip at the top of [this Confluence page](http://simtk-confluence.stanford.edu:8080/pages/viewpage.action?pageId=5113821) that was developed using 3.1 beta, but James may have a newer version.

**Comment by aseth1** [comment]
Thanks @tkuchida. @jimmyDunne is that the model you used and do you see the added Joint when you load the file or only after calling initSystem()?

**Comment by aseth1** [comment]
I was able to reproduce the problem with the model in the download packed. 

The `SimTK::MultibodyGraphMaker`  appears incorrectly handle the Pelvis massless body for the first SliderJoint that yields `Pelvis_tx` dof and ends constructing some weird tree where the thigh bodies are connected to ground via FreeJoints. If I add mass to the first (massless) pelvis body (for first SliderJoint) the correct tree is created and it loads fine. Updating to latest model format did not alleviate the problem. 

I will investigate further, but an immediate alternatives are to: 1) add some mass to the intermediat pelvis body, or 2) use a PlanarJoint instead of two sliders (but it does add rotation DOF to the pelvis though). I think the latter is easier to explain and was what I originally thought the model was using.

**Comment by chrisdembia** [comment]
This also sounds like a bug that should be filed with simbody?

**Comment by jimmyDunne** [comment]
the doxygen describes a planar joint like as

> A PlanarJoint provides three DoFs: rotation about the the common Z of the parent and child joint frames, X and Y translation in the parent body's joint frame.

However, when you copy/paste a default planar joint from the GUI XML Browser, the three coordinates are (1) Translation (2) Rotation (3) Rotation.

**Comment by aseth1** [comment]
The Doxygen is correct for 4.0. Joints were more adhoc in 3.3 and likely to be inconsistent (assuming 3.3 GUI). The default XML for the PlanarJoint should be correct in 4.0 since all Coordinates are now handled by the base class.

**Comment by jimmyDunne** [comment]
Attached is a working planar joint model...

[WalkerModel_planarJoint.osim.zip](https://github.com/opensim-org/opensim-core/files/586547/WalkerModel_planarJoint.osim.zip)

**Comment by jimmyDunne** [comment]
I assume the MultiBodyGraphMaker bug may need its own issue, but this current issue is resolved. Thanks!
